crypto/crypto.go

```diff
@@ -1,128 +1,69 @@
-	secp256k1N, _  = new(big.Int).SetString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141", 16) // nolint: varcheck, staticcheck
-	secp256k1halfN = new(big.Int).Div(secp256k1N, big.NewInt(2))                                                    // nolint: varcheck, staticcheck
-)
+// PrivateKeyBytes is the size of a serialized private key.
+const PrivateKeyBytes = 32
 
-// BytesToECDSA creates a private key with the given D value.
-func BytesToECDSA(d []byte) (*ecdsa.PrivateKey, error) {
-	return toECDSA(d, true)
-}
+// PublicKeyBytes is the size of a serialized public key.
+const PublicKeyBytes = 65
 
-// BytesToECDSAUnsafe blindly converts a binary blob to a private key. It should almost
-// never be used unless you are sure the input is valid and want to avoid hitting
-// errors due to bad origin encoding (0 prefixes cut off).
-func BytesToECDSAUnsafe(d []byte) *ecdsa.PrivateKey {
-	priv, _ := toECDSA(d, false)
-	return priv
+// PublicKey returns the public key for this private key.
+func PublicKey(sk []byte) []byte {
+	x, y := secp256k1.S256().ScalarBaseMult(sk)
+	return elliptic.Marshal(secp256k1.S256(), x, y)
 }
 
-// toECDSA creates a private key with the given D value. The strict parameter
-// controls whether the key's length should be enforced at the curve size or
-// it can also accept legacy encodings (0 prefixes).
-func toECDSA(d []byte, strict bool) (*ecdsa.PrivateKey, error) {
-	priv := new(ecdsa.PrivateKey)
-	priv.PublicKey.Curve = S256()
-	if strict && 8*len(d) != priv.Params().BitSize {
-		return nil, fmt.Errorf("invalid length, need %d bits", priv.Params().BitSize)
-	}
-	priv.D = new(big.Int).SetBytes(d)
-
-	// The priv.D must < N
-	if priv.D.Cmp(secp256k1N) >= 0 {
-		return nil, fmt.Errorf("invalid private key, >=N")
-	}
-	// The priv.D must not be zero or negative.
-	if priv.D.Sign() <= 0 {
-		return nil, fmt.Errorf("invalid private key, zero or negative")
-	}
-
-	priv.PublicKey.X, priv.PublicKey.Y = priv.PublicKey.Curve.ScalarBaseMult(d)
-	if priv.PublicKey.X == nil {
-		return nil, errors.New("invalid private key")
-	}
-	return priv, nil
+// Sign signs the given message, which must be 32 bytes long.
+func Sign(sk, msg []byte) ([]byte, error) {
+	return secp256k1.Sign(msg, sk)
 }
 
-// ECDSAToBytes exports a private key into a binary dump.
-func ECDSAToBytes(priv *ecdsa.PrivateKey) []byte {
-	if priv == nil {
-		return nil
-	}
-	return cu.PaddedBigBytes(priv.D, priv.Params().BitSize/8)
+// Equals compares two private key for equality and returns true if they are the same.
+func Equals(sk, other []byte) bool {
+	return bytes.Equal(sk, other)
 }
 
-// BytesToECDSAPub creates a public key with the given `pub` value.
-func BytesToECDSAPub(pub []byte) *ecdsa.PublicKey {
-	if len(pub) == 0 {
-		return nil
+// Verify checks the given signature and returns true if it is valid.
+func Verify(pk, msg, signature []byte) bool {
+	if len(signature) == 65 {
+		// Drop the V (1byte) in [R | S | V] style signatures.
```

Comment:
in a brief comment can you say something about what this means for the unfamilar? seems weird to accommodate some other format.